modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/features/DocumentationProviderFeatures.java

```diff
@@ -0,0 +1,172 @@
+package org.openapitools.codegen.languages.features;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * The DocumentationProvider Features support to additional properties to select the
+ * documentation provider and the annotation library to use during code generation.
+ *
+ * @author cachescrubber, 2022-01-08
+ * @since 5.4.0
+ */
+public interface DocumentationProviderFeatures {
+
+  String DOCUMENTATION_PROVIDER = "documentationProvider";
+
+  String ANNOTATION_LIBRARY = "annotationLibrary";
+
+  /**
+   * Define the default documentation Provider for CliOpts processing.
+   * Override in subclasses to customize.
+   * @return the default documentation provider
+   */
+  default DocumentationProvider defaultDocumentationProvider() {
+    return DocumentationProvider.NONE;
+  }
+
+  /**
+   * Define the List of supported documentation Provider for CliOpts processing.
+   * Override in subclasses to customize.
+   * @return the list of supported documentation provider
+   */
+  default List<DocumentationProvider> supportedDocumentationProvider() {
+    List<DocumentationProvider> supportedProviders = new ArrayList<>();
+    supportedProviders.add(DocumentationProvider.NONE);
+    return supportedProviders;
+  }
+
+  /**
+   * Define the list of supported annotation libraries for CliOpts processing.
+   * Override in subclasses to customize.
+   * @return the list of supported annotation libraries
+   */
+  default List<AnnotationLibrary> supportedAnnotationLibraries() {
+    List<AnnotationLibrary> supportedLibraries = new ArrayList<>();
+    supportedLibraries.add(AnnotationLibrary.NONE);
+    return supportedLibraries;
+  }
+
+  DocumentationProvider getDocumentationProvider();
+
+  void setDocumentationProvider(DocumentationProvider documentationProvider);
+
+  AnnotationLibrary getAnnotationLibrary();
+
+  void setAnnotationLibrary(AnnotationLibrary annotationLibrary);
+
+  enum DocumentationProvider {
+      NONE("withoutDocumentationProvider", "Do not publish an OpenAPI specification.",
+          AnnotationLibrary.NONE, AnnotationLibrary.values()),
+
+      SOURCE("sourceDocumentationProvider", "Publish the original input OpenAPI specification.",
+          AnnotationLibrary.NONE, AnnotationLibrary.values()),
+
+      SWAGGER1("swagger1DocumentationProvider", "Generate a Swagger 2 specification using Swagger-Core 1.x.",
+          AnnotationLibrary.SWAGGER1, AnnotationLibrary.SWAGGER1),
+
+      SWAGGER2("swagger2DocumentationProvider", "Generate an OpenAPI 3 specification using Swagger-Core 2.x.",
+          AnnotationLibrary.SWAGGER2, AnnotationLibrary.SWAGGER2),
+
```

Comment:
Is the plan to add support for these providers before submitting this PR?

Reply:
No - at least not in the spring generator. General Idea ist that other generators should use the same cli options and template variables to make the project easier to maintain in the long run.

If it is decided to make it spring generator only, I would probably remove those Enum values.